### ISSUE TITLE
 Build ldns failed with SWIG 4.2.0

### ISSUE THREAD

**Issue by jplesnik** [swig/swig#2747]
The build of `ldns-1.8.3` is failing against SWIG 4.2.0 in Fedora rawhide/40.
```
./contrib/python/ldns_wrapper.c: In function '_wrap_ldns_rdf_type2str':
./contrib/python/ldns_wrapper.c:10133:16: error: implicit declaration of function 'SWIG_Python_str_AsChar'; did you mean 'SWIG_Python_str_FromChar'? [-Werror=implicit-function-declaration]
10133 |     arg1_str = SWIG_Python_str_AsChar(swig_obj[0]);
      |                ^~~~~~~~~~~~~~~~~~~~~~
      |                SWIG_Python_str_FromChar
./contrib/python/ldns_wrapper.c:10133:14: warning: assignment to 'char *' from 'int' makes pointer from integer without a cast [-Wint-conversion]
10133 |     arg1_str = SWIG_Python_str_AsChar(swig_obj[0]);
      |              ^
./contrib/python/ldns_wrapper.c:10153:5: error: implicit declaration of function 'SWIG_Python_str_DelForPy3'; did you mean 'SWIG_Python_str_FromFormat'? [-Werror=implicit-function-declaration]
10153 |     SWIG_Python_str_DelForPy3(arg1_str); /* Is a empty macro for Python < 3. */
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~
      |     SWIG_Python_str_FromFormat
```

It should be related to  https://github.com/swig/swig/commit/f89dd59d4b82ece899087682fdb86e94d2611513. 

Complete build log is [here](https://download.copr.fedorainfracloud.org/results/jplesnik/swig-rebuild/fedora-rawhide-x86_64/06843326-ldns/builder-live.log.gz). Other information about the build can be found [here](https://copr.fedorainfracloud.org/coprs/jplesnik/swig-rebuild/build/6843326/).

**Comment by jschueller** [comment]
It looks like this method is not available anymore on purpose and it should be changed in ldns then

**Comment by wsfulton** [comment]
I made this fix, so I'm taking a look at these broken builds and will provide a patch for these open source projects with a fix. Have been grappling with how to reproduce the builds, but have a breakthrough now.

**Comment by wsfulton** [comment]
I've created a merge request with a fix at NLnetLabs/ldns#231. @jplesnik, the build of ldns-1.8.3 works for me with this patch, I suggest you close this issue when you've applied the patch to Fedora.

**Comment by jplesnik** [comment]
@wsfulton Thank you for the fix.
I reported [BZ](https://bugzilla.redhat.com/show_bug.cgi?id=2257901) it to Fedora maintainer of ldns. I hope he applies the fix asap.
